packages/core/query/resourceShapes.rq

```diff
@@ -0,0 +1,24 @@
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+prefix kopflos: <https://kopflos.described.at/>
+
+SELECT DISTINCT ?api ?resourceShape ?subject ?property ?object {
+  ?resourceShape a kopflos:ResourceShape .
+  ?resourceShape kopflos:api ?api .
+
+  {
+    ?resourceShape sh:targetClass ?targetClass .
+    ?type rdfs:subClassOf* ?targetClass .
+    ?subject a ?type .
+  } UNION {
+    ?resourceShape sh:targetNode ?subject .
+  }
+
+  optional {
+    ?resourceShape sh:property/sh:path ?property .
+    ?subject ?property ?object .
+  }
+
+  VALUES ?resource { sh:this }
```

Comment:
I was surprised to find that this `VALUES` only work inside the query and not outside. Thus, the `sh:this` is replaced with the request URL

Reply:
indeed, the query looks non-trivial, and I'm a bit concerned about its speed with large APIs. What about getting rid of `VALUES` and placing `sh:this` directly in all the needed places?

Reply:
Since it's doing replacement, might also do it that way.

Let's address this when it becomes a problem.